doc/langref.html.in

```diff
@@ -1047,6 +972,312 @@ test "init with undefined" {
+    // In this example, the innermost container is this file (implicitly a struct).
+    std.testing.refAllDecls(@This());
+
+    // or, reference each container individually from a top-level test declaration.
+    // The `_ = C;` syntax is a no-op reference to the identifier `C`.
+    _ = S;
+    _ = E;
+    _ = U;
+    _ = O;
+
+    // Alternative to `refAllDecls`, imported source files containing tests can
+    // also be referenced like this:
+    _ = @import("introducing_zig_test.zig");
+}
+
+const S = struct {
+    test "S demo test" {
+        try expect(true);
+    }
+
+    const SE = enum {
+      V,
+
+      // This test won't run because its container (SE) is not referenced.
+      test "This Test Won't Run" {
+          try expect(false);
+      }
+    };
+};
+
+const E = enum {
+    V,
+
+    test "E demo test" {
+        try expect(true);
+    }
+};
+
+const U = union { // U is referenced by the file's top-level test declaration
+    s: US,        // and US is referenced here; therefore, "U.Us demo test" will run
+
+    const US = struct {
+        test "U.US demo test" {
+            // This test is a top-level test declaration for the struct.
+            // The struct is nested (declared) inside of a union.
+            try expect(true);
+        }
+    };
+
+    test "U demo test" {
+        try expect(true);
+    }
+};
+
+const O = opaque {
+    test "O demo test" {
+        try expect(true);
+    }
+};
+      {#code_end#}
+      {#header_close#}
+      {#header_open|Test Failure#}
+      <p>
+        The default test runner checks for an {#link|error|Errors#} returned from a test.
+        When a test returns an error, the test is considered a failure and its {#link|error return trace|Error Return Traces#}
+        is output to standard error. The total number of failures will be reported after all tests have run.
+      </p>
+      {#code_begin|test_err#}
+const std = @import("std");
+
+test "expect this to fail" {
+    try std.testing.expect(false);
+}
+
+test "expect this to succeed" {
+    try std.testing.expect(true);
+}
+      {#code_end#}
+      {#header_close#}
+      {#header_open|Skip Tests#}
+      <p>
+        If a {#syntax#}test{#endsyntax#} returns the error {#syntax#}error.SkipZigTest{#endsyntax#}, the
+        default test runner will consider the test as being skipped. The total number of skipped tests
+        will be reported after all tests have run.
+      </p>
+      {#code_begin|test#}
+test "this will be skipped" {
+    return error.SkipZigTest;
+}
+      {#code_end#}
+      <p>
+        The default test runner also skips tests containing a {#link|suspend point|Async Functions#} and the
+        test is running using the default, blocking IO mode.
+        (The evented IO mode is enabled using the <kbd>--test-evented-io</kbd> command line parameter.)
+      </p>
+      {#code_begin|test|async_skip#}
+const std = @import("std");
+
+// This test is skipped because a suspend point exists
```

Comment:
This comment block can be merged with the description above. Ideally you would also link to the async section to look up what the code exactly does how. Here is simply not enough space to explain all details.